templates/project/src/main.rs

```diff
@@ -89,11 +89,20 @@ fn main() -> anyhow::Result<()> {
         Esp32TlsServerConfig::new(cert, key.as_ptr(), key.len() as u32)
     };
 
+    let mut max_connection = 3;
+    unsafe {
+        if !g_spiram_ok {
+            log::info!("spiram not initialized disabling cache feature of the wifi driver");
+            g_wifi_feature_caps &= !(CONFIG_FEATURE_CACHE_TX_BUF_BIT as u64);
+            max_connection = 1;
+        }
```

Comment:
Taken from examples. 

 Could also just hardcode?

If you would rather do something else, feel free to close the PR and implement yourself